### PR TITLE
Rewrite array operator to function in parser

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

Part of #8506 and #10534
and nice to have for #9857

### CODE REVIEW

File: datafusion/functions-array/src/rewrite.rs [hunk 1]
```diff
@@ -61,91 +59,6 @@ impl FunctionRewrite for ArrayFunctionRewriter {
                 Transformed::yes(array_has_all(*right, *left))
             }
 
-            // Column cases:
```
Comment (alamb) [inline datafusion/functions-array/src/rewrite.rs:64]: Should we do the same thing for the other operators (AtArrow)?

**Review by alamb — APPROVED** [review]
Looks like a nice cleanup to me . Thank you @jayzhan211

File: datafusion/functions-array/src/rewrite.rs (hunk 1 above)
Comment (jayzhan211) [reply datafusion/functions-array/src/rewrite.rs:64]: yes!

**Comment by jayzhan211** [comment]
Thanks @alamb

**Pull request merged**